drizzle-orm/src/mysql-core/functions/aggregate.ts

```diff
@@ -0,0 +1,149 @@
+import { is, entityKind } from '~/entity.ts';
+import { MySqlColumn } from '../columns/index.ts';
+import { sql, type SQLWrapper, isSQLWrapper, type SQLChunk } from '~/sql/index.ts';
+import { MySqlBuiltInFunction } from './common.ts';
+import { type MaybeDistinct, getValueWithDistinct } from '~/distinct.ts';
+
+export class MySqlAggregateFunction<T = unknown> extends MySqlBuiltInFunction<T> {
+	static readonly [entityKind]: string = 'MySqlAggregateFunction';
+}
+
+/**
+ * Returns the number of values in `expression`.
+ *
+ * ## Examples
+ *
+ * ```ts
+ * // Number employees with null values
+ * db.select({ value: count() }).from(employees)
+ * // Number of employees where `name` is not null
+ * db.select({ value: count(employees.name) }).from(employees)
+ * // Number of employees where `name` is distinct (no duplicates)
+ * db.select({ value: count(distinct(employees.name)) }).from(employees)
+ * ```
+ */
+export function count<T extends 'number' | 'bigint' | undefined = undefined>(expression?: MaybeDistinct<SQLWrapper> | '*', config?: {
+	mode: T;
+}): MySqlAggregateFunction<T extends 'number' ? number : bigint> {
+	const { value, distinct } = getValueWithDistinct(expression);
+	const chunks: SQLChunk[] = [];
+
+	if (distinct) {
+		chunks.push(sql`distinct `);
+	}
+	chunks.push(isSQLWrapper(value) ? value : sql`*`);
+
+	const sql_ = sql
+		.join([sql`count(`, ...chunks, sql`)` ])
+		.mapWith(config?.mode === 'number' ? Number : BigInt);
+
+	return new MySqlAggregateFunction(sql_) as any;
+}
+
+/**
+ * Returns the average (arithmetic mean) of all non-null values in `expression`.
+ *
+ * ## Examples
+ *
+ * ```ts
+ * // Average salary of an employee
+ * db.select({ value: avg(employees.salary) }).from(employees)
+ * // Average salary of an employee where `salary` is distinct (no duplicates)
+ * db.select({ value: avg(distinct(employees.salary)) }).from(employees)
+ * ```
+ */
+export function avg<T extends 'number' | 'bigint' | 'string' | undefined = undefined>(expression: MaybeDistinct<SQLWrapper>, config?: {
```

Comment:
It seems to me like the implementation of all this functions are dialect agnostic?
I think it would be better to implement all common functions outside the dialect-specific folder. for example in `~/sql/functions.ts`

Reply:
Updated to be implemented as such.